apps/gitness/src/pages-v2/repo/repo-sidebar.tsx

```diff
@@ -23,24 +23,33 @@ import { useRepoBranchesStore } from '././stores/repo-branches-store.ts'
  * TODO: This code was migrated from V2 and needs to be refactored.
  */
 export const RepoSidebar = () => {
-  const { selectedBranchTag, setSelectedBranchTag, selectedBranchType, setSpaceIdAndRepoId } = useRepoBranchesStore()
+  const {
+    branchList,
+    tagList,
+    setBranchList,
+    setTagList,
+    selectedBranchTag,
+    setSelectedBranchTag,
+    setSelectedBranchType,
+    setSpaceIdAndRepoId
+  } = useRepoBranchesStore()
 
   const repoRef = useGetRepoRef()
   const { spaceId, repoId } = useParams<PathParams>()
   const { fullGitRef, gitRefName, fullResourcePath } = useCodePathDetails()
   const navigate = useNavigate()
 
   useEffect(() => {
-    !selectedBranchTag && setSelectedBranchTag({ name: fullGitRef || '', sha: '' })
+    !selectedBranchTag && setSelectedBranchTag({ name: gitRefName || '', sha: '' })
```

Comment:
Do we really need !selectedBranchTag condition? Or can we directly set the state?

Reply:
If we need the condition, we should probably add it in the setter method in store.

Reply:
yeah it can be directly set, fullGitRef from route would be the source of truth